zerver/views/invite.py

```diff
@@ -76,6 +76,9 @@ def revoke_user_invite(request: HttpRequest, user_profile: UserProfile,
     if prereg_user.referred_by.realm != user_profile.realm:
         raise JsonableError(_("No such invitation"))
 
+    if prereg_user.referred_by_id != user_profile.id and not user_profile.is_realm_admin:
+        raise JsonableError(_("You can't revoke invitations sent by others"))
```

Comment:
We should add a test for this case, as well as the one below.  And I think we should use the generic "Must be an administrator" error that `require_realm_admin` would have used; there's no reason to force translators to translate an extra string for this.

`InvitationsTestCase` is the place to put these.

Reply:
```
diff --git a/zerver/lib/actions.py b/zerver/lib/actions.py
index 79b0387acc..4d9dd2b3fe 100644
--- a/zerver/lib/actions.py
+++ b/zerver/lib/actions.py
@@ -5099,12 +5099,14 @@ def do_invite_users(user_profile: UserProfile,
 def do_get_user_invites(user_profile: UserProfile) -> List[Dict[str, Any]]:
     days_to_activate = settings.INVITATION_LINK_VALIDITY_DAYS
     active_value = getattr(confirmation_settings, 'STATUS_ACTIVE', 1)
-
     lowest_datetime = timezone_now() - datetime.timedelta(days=days_to_activate)
-    prereg_users = PreregistrationUser.objects.exclude(status=active_value).filter(
+    base_query = PreregistrationUser.objects.exclude(status=active_value).filter(
         invited_at__gte=lowest_datetime)
-    if not user_profile.is_realm_admin:
-        prereg_users = prereg_users.filter(referred_by=user_profile)
+
+    if user_profile.is_realm_admin:
+        prereg_users = base_query.filter(referred_by__realm=user_profile.realm)
+    else:
+        prereg_users = base_query.filter(referred_by=user_profile)
 
     invites = []
 
diff --git a/zerver/tests/test_signup.py b/zerver/tests/test_signup.py
index 6dae0145e4..61e517da7a 100644
--- a/zerver/tests/test_signup.py
+++ b/zerver/tests/test_signup.py
@@ -1483,6 +1483,9 @@ class InvitationsTestCase(InviteUserBase):
         prereg_user_three.save()
         prereg_user_four = PreregistrationUser(email="TestFour@zulip.com", referred_by=othello)
         prereg_user_four.save()
+        prereg_user_other_realm = PreregistrationUser(
+            email="TestOne@zulip.com", referred_by=self.mit_user("sipbtest"))
+        prereg_user_other_realm.save()
         self.assertEqual(len(do_get_user_invites(user_profile)), 4)
         self.assertEqual(len(do_get_user_invites(hamlet)), 1)
         self.assertEqual(len(do_get_user_invites(othello)), 1)
```